### DISCUSSION TITLE
Just Ask

### DISCUSSION THREAD

**Discussion by vaguecoder** [vaguecoder/gorilla-mux#23] [General]
Hi,
This is an open forum. Just ask what you wanna ask here about this forked project.

**Comment by Rishav1707** [comment]
By mistake, I just squash the last 5 commits.

**Comment by Rishav1707** [comment]
How to undo these commits, as it leads to some unnecessary commits which do not belong to my changes.

**Reply by vaguecoder** [reply]
If those changes are only in local and remote branch still has the correct commits, just discard the existing changes in local and do a pull
- Check uncommitted changes `git status`.
- If status has staged changes, do `git restore --staged .`
- If status has uncommitted changes, do `git checkout -- .`
- After doing the above two, if there are still files uncommitted, they could be untracked files. Delete them manually.
- Check the status again. If all clear, it it better to jump on to an older commit before doing a pull. Do `git reset --hard HEAD~10`.
- If things come to square one, you know how to squash 4 commits.

Contrastly, if you don't have the actual 4 commits intact in both local and remote branch, don't do the above steps as you may lose the changes. Do the following:
- Do `git status`. If there are uncommitted changes, go to next step. Else if the changes are all in last commit, do `git reset --soft HEAD~1` and go to next step.
- Do `git stash --include-untracked` to stash all untracked changes (including HEAD-5th commit's changes too). Let it be.
- Hoping your master has the latest changes from this repo's master, do a `git rebase master`. You should now see the lost previous commit in `git log` that you accidentally squashed earlier.
- Now do `git stash pop` to see uncommitted changes. If you see changes more than what you've added, delete the changes manually.
- Else, commit and force push.

So much to try but unfortunately, there are no easy ways unless you know what you have to do.
